miner/worker.go

```diff
@@ -603,3 +624,23 @@ func (miner *Miner) validateParams(genParams *generateParams) (time.Duration, er
 	}
 	return time.Duration(blockTime) * time.Second, nil
 }
+
+func (miner *Miner) blockFeeCurrency(env *environment, tx *types.Transaction, err error) {
+	if tx.FeeCurrency() != nil {
```

Comment:
Feels like we can avoid this check if we change the signature of the function to accept the currency as all the conditions are actually checked by the caller already.